programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -17,107 +18,101 @@
-            } else {
-                ZITI_LOG(WARN, "ziti_ctx controller connections failed: %s", ziti_errorstr(event->event.ctx.ctrl_status));
-            }
-            break;
+static void on_cmd(uv_stream_t *s, ssize_t len, const uv_buf_t *b) {
+    if (len < 0) {
+        uv_close((uv_handle_t *) s, NULL);
+    } else {
+        ZITI_LOG(INFO, "received cmd <%.*s>", (int) len, b->base);
 
-        case ZitiServiceEvent: {
-            ziti_service **zs;
-            for (zs = event->event.service.removed; *zs != NULL; zs++) {
-                on_service(ztx, *zs, ZITI_SERVICE_UNAVAILABLE, ziti_app_ctx(ztx));
-            }
-            for (zs = event->event.service.added; *zs != NULL; zs++) {
-                on_service(ztx, *zs, ZITI_OK, ziti_app_ctx(ztx));
-            }
-            for (zs = event->event.service.changed; *zs != NULL; zs++) {
-                on_service(ztx, *zs, ZITI_OK, ziti_app_ctx(ztx));
-            }
-            break;
+        tunnel_comand cmd = {0};
+        if (parse_tunnel_comand(&cmd, b->base, len) == 0) {
+            CMD_CTRL->process(&cmd, on_command_resp, NULL);
+        } else {
+            tunnel_result resp = {
+                    .success = false,
+                    .error = "failed to parse command"
+            };
+            on_command_resp(&resp, NULL);
         }
+        free_tunnel_comand(&cmd);
+    }
+
+    free(b->base);
+}
 
-        case ZitiRouterEvent:
-            break;
+static void on_cmd_client(uv_stream_t *s, int status) {
+    // only allow a single client
+    if (!uv_is_active((const uv_handle_t *) &cmd_conn)) {
+        uv_pipe_init(s->loop, &cmd_conn, 0);
+        uv_accept(s, (uv_stream_t *) &cmd_conn);
+        uv_read_start((uv_stream_t *) &cmd_conn, cmd_alloc, on_cmd);
+    } else {
+        uv_pipe_t *tmp = malloc(sizeof(uv_pipe_t));
+        uv_pipe_init(s->loop, tmp, 0);
+        uv_accept(s, (uv_stream_t *) tmp);
+        uv_close((uv_handle_t *) tmp, (uv_close_cb) free);
     }
 }
 
-static void load_ziti_async(uv_async_t *ar) {
-    struct ziti_instance_s *inst = ar->data;
+static int start_cmd_socket(uv_loop_t *l) {
 
-    char *config_path = realpath(inst->opts.config, NULL);
-    if (model_map_get(&instances, config_path) != NULL) {
-        ZITI_LOG(WARN, "ziti context already loaded for %s", inst->opts.config);
-    } else {
-        ZITI_LOG(INFO, "loading ziti instance from %s", config_path);
-        inst->opts.app_ctx = tnlr_ctx;
-        if (ziti_init_opts(&inst->opts, ar->loop) == ZITI_OK) {
-            model_map_set(&instances, config_path, inst);
-        } else {
-            ZITI_LOG(ERROR, "failed to initialize ziti");
-        }
+    if (uv_is_active((const uv_handle_t *) &cmd_server)) {
+        return 0;
     }
-    free(config_path);
-    uv_close((uv_handle_t *) ar, (uv_close_cb) free);
+    static char sockfile[] = "/tmp/ziti-edge-tunnel.sock";
```

Comment:
I know it isn't something that we want to do very often, but it is currently possible to run multiple ziti-edge-tunnels on the same host. This might be more useful if/when we support host mode... Anyway, what do you think about putting the PID in the socket file name?

Reply:
instead of pid tun name would be better I think -- `/tmp/ziti-edge-tunnel.tun0.sock`
we'll do it later when we know how UI can deal with it

Reply:
I thought about tun name, but there won't be a tun device in some cases (e.g. if we support tproxy or something like host-only mode).